pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DataTypeConversionFunctions.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.function.scalar;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import org.apache.pinot.common.function.annotations.ScalarFunction;
+
+
+public class DataTypeConversionFunctions {
+  private DataTypeConversionFunctions() {
+
+  }
+
+  @ScalarFunction
```

Comment:
- What is the format here? Are we saying that byte array is the 2's complement representation of the underlying unscaled value. 
- How is scale represented in the byte array?
- Do we have to take care of endianness here? The unscaled values is going to be BIG-ENDIAN. How's it working without swapping bytes?

Reply:
* Yes, it is the 2's complement representation of underling unscaled value

* Scale is the first 4 bytes in the byte array, the remaining bytes are of unscaled value

* The unscaled value byte array is indeed Big-endian and they are copied as such in the result array. There's no need to swap the bytes.

In summary, the final byte array is just unscaled value's bytes appended to scale's bytes

Reply:
@KKcorps , can we add some e2e query execution functional tests?

Reply:
@siddharthteotia Yes, the tests are almost done. Will push them tomorrow.

Reply:
@siddharthteotia Tests have been added. Let me know if any changes are needed.